### ISSUE TITLE
Intermittent error when calling db.prepare(stmt)

### ISSUE THREAD

**Issue by amayne** [stephencelis/SQLite.swift#223]
I am getting this error intermittently (every 3 or 4 times) when selecting from the database. I am using the latest from the swift-2 branch: 
```swift
fatal error: 'try!' expression unexpectedly raised an error: authorization denied: file /Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-700.0.59/src/swift/stdlib/public/core/ErrorType.swift, line 50
```

My code looks like:
```swift
func getDataWithName(nm: String) -> Data? {
        return resolveClosestMatch(data.filter(name == nm)).join(groups, on: groupId == group[id]).order(version.desc).limit(2))
    }

func resolveClosestMatch(query: QueryType) -> Data? {
        var bestData: String? = nil
        for dataRow in db.prepare(query) {
            ...
        }
        return bestData
    }
```

I realize this branch has not been merged yet, but would love this to be resolved! Unsure why I am getting this error

**Comment by stephencelis** [comment]
@amayne I'll likely be merging in #221 soon, which should allow you to recover from prepare issues. As for the exact error, this happens (I think) when the device is locked and sqlite3 is trying to access the db file. Is your code running while the device is locked?

**Comment by amayne** [comment]
Thanks @stephencelis ! I do use SQLite in background refresh (while the device is locked), but this error I'm seeing seems to happen on launch (LaunchScreen nib is showing) and a controller's `viewDidLoad-` is querying the SQLite database. Is that expected? This seems to have cropped up with iOS9/Swift 2.

**Comment by mikemee** [comment]
Perhaps fixed by the accepted PR #290 - at least you should be able to catch it in your code and see the error. Please re-open if this is still a problem after updating with the latest release. Thanks!

**Comment by JoeriBultheel** [comment]
Hi, i'm getting this error when clicking on a remote notifications action. So yeah i want the database to be accessed while the device is locked. Is this not possible using SQLite.swift? I need a solution fast! Thanks
